github/compliance/report.py

```diff
@@ -300,31 +397,25 @@ def _latest_processing_date(
     )
 
 
-def target_from_component_artifact(
-    component: cm.Component,
-    artifact: cm.Resource | cm.ComponentSource,
-) -> cnudie.iter.ResourceNode | cnudie.iter.SourceNode:
-    path = (component,)
-
-    if isinstance(artifact, cm.ComponentSource):
-        return cnudie.iter.SourceNode(
-            path=path,
-            source=artifact,
-        )
-    if isinstance(artifact, cm.Resource):
-        return cnudie.iter.ResourceNode(
-            path=path,
-            resource=artifact,
-        )
-
-    raise RuntimeError(f'{artifact=}')
-
-
 class PROCESSING_ACTION(enum.Enum):
     DISCARD = 'discard'
     REPORT = 'report'
 
 
+def _issue_type_close_comment(issue_type: str) -> str:
+    if issue_type in (
+        _compliance_label_vulnerabilities,
+        _compliance_label_licenses,
+        _compliance_label_os_outdated,
+        _compliance_label_checkmarx,
+        _compliance_label_malware,
+    ):
+        return 'closing, because component/resource no longer present in BoM'
+
+    else:
+        raise NotImplementedError(f'{issue_type=}')
```

Comment:
I think, during our discussion on builtin exceptions, you found out (and convinced me) that NIE should not be used here (instead, VE should be more adequate)..

Reply:
We did agree on NIE for iterating over Enum. Of course, this is not an Enum but it feels quite similar. But no strong opinion of mine.

Reply:
I am fine w/ NIE, considering we use it all over our codebase already.

however, please do not unnecessarily wrap values. in this case, we will even lose information as `__str__`  will be called, which most likely will yield less output.

I cannot stress this (or so it seems) often enough: please just pass the offensive values _unaltered_ to builtin exceptions, such as ValueError, TypeError, NotImpementedError,. ..

If - and _only if_ - there is more specific (and accurate (!) information to be added that will be helpful (!) for someone reading the exception-message - then, and only then, add an error-text.

in this specific case (we use this pattern a lot), the only reason why this exception will ever be raised will be because either we did a programming error and passed-in a completely wrong value, or because we extended to allowed set of values (/types), but did forgot this one location. in which case the stacktrace + full dump about the value is exactly what we need. anything else will actually make it harder to quickly identify what the underlying issue is.